backend/infrahub/core/manager.py

```diff
@@ -1136,33 +1138,37 @@ async def get_many(  # pylint: disable=too-many-branches,too-many-statements
             profile_attributes_id_map=profile_attributes, profile_ids_by_node_id=profile_ids_by_node_id
         )
 
-        # if prefetch_relationships is enabled
+        has_relationship_fields = False
+        if fields and nodes_info_by_id:
+            for node_info in nodes_info_by_id.values():
+                if node_info.schema and set(fields.keys()) & set(node_info.schema.relationship_names):
+                    has_relationship_fields = True
+                    break
+
+        # if prefetch_relationships or fields include relationships
         # Query all the peers associated with all nodes at once.
-        peers_per_node = None
-        peers = None
-        if prefetch_relationships:
+        grouped_peer_nodes: GroupedPeerNodes | None = None
+        peers: dict[str, Node] = {}
+        if prefetch_relationships or has_relationship_fields:
```

Comment:
The query `NodeListGetRelationshipsQuery` needs to have some filters to only query the relatiionships we really care about. Currently, it will return all peers of all relationships which could lead to some pretty bad performance

Reply:
added the filter for relationship identifiers to the query and refactored `get_many` to use it